assessments/projects/cni/self-assessment.md

```diff
@@ -0,0 +1,249 @@
+
+![Overview of how CNI fits into cloud native world](docs/CNI-role.png)
+
+### Actors
+
+The following are the actors found in CNI project:
+* **CNI Plugin**: A program that applies a specified network configuration. It interfaces with a variety of networking solutions to provide flexible networking capabilities.
+* **Container**: A network isolation domain, though the actual isolation technology is not defined by the specification.
+* **Container Runtime**: Is responsible for executing CNI plugins. It interacts with plugins to apply network configurations to individual containers.
+* **Network**: A group of endpoints that are uniquely addressable that can communicate amongst each other. This could be either an individual container (as specified above), a machine, or some other network device (e.g. a router). Containers can be conceptually added to or removed from one or more networks.It includes configuration data such as IP addresses, routing rules, and network policies.
+
+### Actions
+
+#### Action Overview
+
+1. A format for administrators defines network configurations
+2. A protocol for container runtimes makes requests to network plugins with parameters from environment variables and network configuration
+3. With the supplied configuration, a procedure executes the plugins.
+4. CNI plugins may also have a procedure to delegate functionality to other plugins
+5. Data types for plugins return their results to the runtime
+
+#### Action 1: Network Configuration Format
+
+CNI defines a network configuration format for administrators. It contains directives for both the container runtime as well as the plugins
+to consume. At plugin execution time, this configuration format is interpreted by the runtime and transformed into a form to be passed to the plugins.
+* **Example configuration**:
+  <br>
+  <div style="text-align:center;">
+    <img src="docs/CNI-config.png" alt="CNI config">
+  </div>
+  <br>
+
+#### Action 2: Execution Protocol
+
+The CNI protocol is based on execution of binaries invoked by the container runtime. CNI defines the protocol between the plugin binary and the runtime. CNI defines 4 operations: ADD, DEL, CHECK, and VERSION. These are passed to the plugin via the CNI\_COMMAND environment variable.
+
+#### Action 3: Execution of Network Configurations
+
+A container time interprets a network configuration and executes plugins accordingly. A runtime can add, delete, or check a network configuration in a container, which results in a series of plugin ADD, DELETE, or CHECK executions correspondingly.
+
+#### Action 4: Plugin Delegation
+
+There are some operations that, for whatever reason, cannot reasonably be implemented as a discrete chained plugin. Rather, a CNI plugin may wish to delegate some functionality to another plugin. One common example of this is IP address management.
+
+#### Action 5: Result Types
+
+Plugins can return one of three result types:
+* Success
+* Error
+* \_Version
```

Comment:
I wish we had more context here about why these are listed and what the meaning of these is in context...